File: atcoder/abc393b_a_b_c.zig
// Vicfred
// https://atcoder.jp/contests/abc393/tasks/abc393_a
// implementation

const std = @import("std");

fn countABCTriplets(s: []const u8) u32 {
    var triples: u32 = 0;
    const n = s.len;
    for (0..n) |i| {
        for (i + 1..n) |j| {
            for (j + 1..n) |k| {
                if (j - i != k - j) {
                    continue;
                }
                if (s[i] == 'A' and s[j] == 'B' and s[k] == 'C') {
                    triples += 1;
                }
            }
        }
    }
    return triples;
}

pub fn main() !void {
    var buffer: [100]u8 = undefined;
    const stdin = std.io.getStdIn();
    const line = try stdin.reader().readUntilDelimiterOrEof(&buffer, '\n');

    if (line) |input| {
        const trimmed = std.mem.trim(u8, input, " \t\n");
        std.debug.print("{}\n", .{countABCTriplets(trimmed)});
    } else {
        std.debug.print("Invalid input\n", .{});
    }
}

File: atcoder/abc393b_a_b_c.rs
// Vicfred
// https://atcoder.jp/contests/abc393/tasks/abc393_a
// implementation

use std::io;

fn count_abc_triplets(s: &str) -> i32 {
    let n = s.len();
    let mut triples = 0;
    
    for i in 0..n {
        for j in i + 1..n {
            for k in j + 1..n {
                if j - i != k - j {
                    continue;
                }
                if s.chars().nth(i) == Some('A') && s.chars().nth(j) == Some('B') && s.chars().nth(k) == Some('C') {
                    triples += 1;
                }
            }
        }
    }
    triples
}

fn main() {
    // Read input string
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let input = input.trim();
    println!("{}", count_abc_triplets(input));
}
